pyanaconda/modules/storage/devicetree/utils.py

```diff
@@ -290,3 +290,24 @@ def find_unconfigured_luks(storage):
         devices.append(device)
 
     return devices
+
+def resolve_device(storage, dev_spec):
+    """Return the device matching the provided device specification.
+
+    A wrapper of Blivet's resolve_device function supporting also
+    resolution of btrfs volumes specified by UUID of the volume
+    and name of the subvolume, as:
+    'UUID=2252ec30-1fce-4f8e-bdef-c50c3a44ede4@root'
+
+    :param storage: an instance of Blivet
+    :param dev_spec: a string describing a block device
+    :returns: the device
+    :rtype: :class:`~.devices.StorageDevice` or None
+    """
+
+    options = None
+    if dev_spec.startswith("UUID") and "@" in dev_spec:
```

Comment:
@vojtechtrefny Couldn't this be part of Blivet or is it too Anaconda specific?

Reply:
I think we could add an extra argument for `resolve_device` in blivet, something like `subvolspec` just for btrfs.

Reply:
I am not sure we want to add it also to https://github.com/rvykydal/anaconda/blob/31db7b8e5f6b423ea8a3ec2663b42c0720a771e9/pyanaconda/modules/storage/devicetree/viewer_interface.py#LL104C7-L104C7. I am not sure if we want to diverge here though (device specificaiton for Anaconda's ResolveDevice and blivet's resolve_device). See also my other comment about adding a special device specification attribute to MountPointRequest.
@poncovka what do you think?